pymc3/sampling.py

```diff
@@ -132,7 +142,14 @@ def sample(draws, step=None, start=None, trace=None, chain=0, njobs=1, tune=None
     """
     model = modelcontext(model)
 
-    step = assign_step_methods(model, step)
+    if step is None and init is not None and pm.model.all_continuous(model.vars):
```

Comment:
Do we really need to check `pm.model.all_continuous(model.vars)`?
Suppose `init=='map'` that does not require `pm.model.all_continuous(model.vars)` to be true then it will be rejected according to this if statement. There should be something like switching default: primary is advi, secondary is map